silabs-pti/src/main/java/com/silabs/pti/Main.java

```diff
@@ -251,6 +271,24 @@ private <T> int runCapture(final IDebugChannelExportFormat<T> format,
     return 0;
   }
 
+  /**
+   *  Parse IP and return <address,port> pair
+   * @param ip ip address to parse
+   * @return when IP has format address:port, return <address,port> pair. when IP
+   * does not have port or is invalid value (e.g. null or empty value), return null.
+   */
+  private Map.Entry<String, Integer> parseIpAddrAndPort(String ip) {
```

Comment:
I don't like the use of `Map.Entry` here. It would be much cleaner to simply create a new object: AddressAndPort, which I mentioned before.

In my opinion, creating specific-need objects always results in a cleaner java code, than attempting to reuse some java library objects, which inevitably tries to fit a square peg into a round hole.